### PR TITLE
Update SentimentAnalysis tutorial to 1.0-preview

### LINKED ISSUE #11779: Tutorials document requires an update for ML version 1.0.0-preview
The following three changes need to be required to work Price Prediction Tutorial work correctly,

#01. Install-Package Microsoft.ML.FastTree after creating the console application
On Create a console application, another ML package "Microsoft.ML.FastTree" need to be installed.

&gt;&gt; dotnet add package Microsoft.ML.FastTree 

#02. Save() function in SaveModelAsFile method will have an extra argument as null,

&gt;&gt;mlContext.Model.Save(model, fileStream)  to  mlContext.Model.Save(model, null, fileStream)

#03. Load() function in Method TestSinglePrediction will have extra argument

            ITransformer loadedModel; 
            Microsoft.ML.DataViewSchema inputSchema;          
            using (var stream = new FileStream(_modelPath, FileMode.Open, FileAccess.Read, FileShare.Read))
            {
                loadedModel = mlContext.Model.Load(stream,out inputSchema);
            } 

End

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: a30b3ff5-b5e6-ff02-6f60-b5cef6f34666
* Version Independent ID: fdf962e2-620e-d9ad-5057-5f96f52dc60f
* Content: [Predict prices using a regression learner with ML.NET](https://docs.microsoft.com/en-us/dotnet/machine-learning/tutorials/taxi-fare#create-a-console-application)
* Content Source: [docs/machine-learning/tutorials/taxi-fare.md](https://github.com/dotnet/docs/blob/master/docs/machine-learning/tutorials/taxi-fare.md)
* Product: **dotnet-ml**
* GitHub Login: @aditidugar
* Microsoft Alias: **johalex**

### LINKED ISSUE #11859: Tutorial doesn't work in 1.0RC
The StochasticDualCoordinateAscent is not available in the RC. I was able to follow the tutorial, but going to version 0.10; and it works with 0.11.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 77fc5686-2288-d607-8136-3146ed02dbf6
* Version Independent ID: 8a16b9e9-b5cb-a55e-2135-191bc62c1656
* Content: [Use ML.NET in a GitHub issue multiclass classification scenario](https://docs.microsoft.com/en-us/dotnet/machine-learning/tutorials/github-issue-classification#load-the-data)
* Content Source: [docs/machine-learning/tutorials/github-issue-classification.md](https://github.com/dotnet/docs/blob/master/docs/machine-learning/tutorials/github-issue-classification.md)
* Product: **dotnet-ml**
* GitHub Login: @JRAlexander
* Microsoft Alias: **johalex**

### PR DESCRIPTION
**Pull request by JRAlexander**
[Internal Review URL](https://review.docs.microsoft.com/en-us/dotnet/machine-learning/tutorials/sentiment-analysis?branch=pr-en-us-11816)

Fixes https://github.com/dotnet/docs/issues/11859
Fixes https://github.com/dotnet/docs/issues/11779

Code here: https://github.com/dotnet/samples/pull/81

### CODE REVIEW

**Review by natke — CHANGES_REQUESTED** [review]
Some general comments:
- the Build and Train Model section doesn't actually build and train the model - it more sets up skeleton functions
- In the closing para, you mentioned hyper-parameters but we've not defined these yet
- There is some confusion between the machine learning task and the algorithm to choose for that task. (And I know we don't provide good guidance on that rn). It would be good to clarify this, even if we remain a little vague about why we chose the Scda algorithm.
- I'm not sure we need to save and re-load the model in every single tutorial. This seems like an operational feature that developers could learn about from our How-to on the topic.

File: docs/machine-learning/tutorials/sentiment-analysis.md [hunk 1]
```diff
@@ -168,33 +118,32 @@ The `LoadData` method executes the following tasks:
...
-<xref:Microsoft.Data.DataView.IDataView>.
-
-As the input and output of `Transforms`, a `DataView` is the fundamental data pipeline type, comparable to `IEnumerable` for `LINQ`.
-
-In ML.NET, data is similar to a SQL view. It is lazily evaluated, schematized, and heterogenous. The object is the first part of the pipeline, and loads the data. For this tutorial, it loads a dataset with comments and corresponding toxic or non toxic sentiment. This is used to create the model, and train it.
+There are three major concepts in ML.NET: [Data](../basic-concepts-model-training-in-mldotnet.md#data), [Transformers](../basic-concepts-model-training-in-mldotnet.md#transformer), and [Estimators](../basic-concepts-model-training-in-mldotnet.md#estimator).
```
[the remark this answers is not in the document]
Comment (JRAlexander) [reply docs/machine-learning/tutorials/sentiment-analysis.md]: We need to define them somewhere.

File: docs/machine-learning/tutorials/sentiment-analysis.md [hunk 2]
```diff
@@ -221,48 +170,62 @@ public static ITransformer BuildAndTrainModel(MLContext mlContext, IDataView spl
...
 
-[!code-csharp[FeaturizeText](~/samples/machine-learning/tutorials/SentimentAnalysis/Program.cs#FeaturizeText "Featurize the text")]
+|SentimentText                         |Sentiment |Features              |
+|--------------------------------------|----------|----------------------|
+|Waitress was a little slow in service.|    0     |[0.76, 0.65, 0.44, …] |
+|Crust is not good.                    |    0     |[0.98, 0.43, 0.54, …] |
+|Wow... Loved this place.              |    1     |[0.35, 0.73, 0.46, …] |
+|Service was very prompt.              |    1     |[0.39, 0, 0.75, …]    |
 
 >[!WARNING]
```
[the remark this answers is not in the document]
Comment (JRAlexander) [reply docs/machine-learning/tutorials/sentiment-analysis.md]: Nope!

File: docs/machine-learning/tutorials/sentiment-analysis.md [hunk 3]
```diff
@@ -221,48 +170,62 @@ public static ITransformer BuildAndTrainModel(MLContext mlContext, IDataView spl
...
+## Add a learning algorithm
 
-## Choose a learning algorithm
+### About the classification task
 
-To add the trainer, call the `mlContext.BinaryClassification.Trainers.FastTree` wrapper method which returns a <xref:Microsoft.ML.Trainers.FastTree.FastTreeBinaryClassificationTrainer> object. This is a decision tree learner you'll use in this pipeline. The `FastTreeBinaryClassificationTrainer` is appended to the `pipeline` and accepts the featurized `SentimentText` (`Features`) and the `Label` input parameters to learn from the historic data.
+"Is it A or B?"
 
-Add the following code to the `BuildAndTrainModel` method:
+![classification machine learning algorithm](./media/sentiment-analysis/classification.png)
```
[the remark this answers is not in the document]
Comment (JRAlexander) [reply docs/machine-learning/tutorials/sentiment-analysis.md]: Sure. Can we leave it for now in the meantime and replace later?

File: docs/machine-learning/tutorials/sentiment-analysis.md [hunk 4]
```diff
@@ -221,48 +170,62 @@ public static ITransformer BuildAndTrainModel(MLContext mlContext, IDataView spl
...
 
-Add the following code to the `BuildAndTrainModel` method:
+## Train the model
+
+Fit the model to the `Train` data and return the trained model by adding the following as the next line of code in the `BuildAndTrainModel()` method:
 
 [!code-csharp[TrainModel](~/samples/machine-learning/tutorials/SentimentAnalysis/Program.cs#TrainModel "Train the model")]
 
-### Save and Return the model trained to use for evaluation
+The [Fit()](xref:Microsoft.ML.Trainers.MatrixFactorizationTrainer.Fit%28Microsoft.ML.IDataView,Microsoft.ML.IDataView%29) method trains your model with the provided training dataset. It executes the `Estimator` definitions by transforming the data and applying the training, and it returns back the trained model, which is a `Transformer`.
```
[the remark this answers is not in the document]
Comment (JRAlexander) [reply docs/machine-learning/tutorials/sentiment-analysis.md]: Not sure that is clearer

File: docs/machine-learning/tutorials/sentiment-analysis.md [hunk 5]
```diff
@@ -274,52 +237,39 @@ public static void Evaluate(MLContext mlContext, ITransformer model, IDataView s
...
-private static void SaveModelAsFile(MLContext mlContext, ITransformer model)
-{
-
-}
-```
-
-The `SaveModelAsFile` method executes the following tasks:
-
-* Saves the model as a .zip file.
+## Save model as .zip file
```
[the remark this answers is not in the document]
Comment (JRAlexander) [reply docs/machine-learning/tutorials/sentiment-analysis.md]: agreed.

File: docs/machine-learning/tutorials/sentiment-analysis.md [hunk 6]
```diff
@@ -145,13 +84,22 @@ Remove the existing class definition and add the following code, which has two c
...
+
+|SentimentText                         |Sentiment (Label) |
+|--------------------------------------|----------|
+|Waitress was a little slow in service.|    0     |
+|Crust is not good.                    |    0     |
+|Wow... Loved this place.              |    1     |
+|Service was very prompt.              |    1     |
+
+`SentimentPrediction` is the prediction class used after the model training. It has a single boolean (`Sentiment`) and a `PredictedLabel` `ColumnName` attribute. The `Label` is used to create and train the model, and it's also used with the split out test dataset to evaluate the model. The `PredictedLabel` is used during prediction and evaluation. For evaluation, training data, the predicted values, and the model are used.
```
Comment (luisquintanilla) [inline docs/machine-learning/tutorials/sentiment-analysis.md]: Suggestion: Add xref to `ColumnName`.

File: docs/machine-learning/tutorials/sentiment-analysis.md [hunk 7]
```diff
@@ -221,48 +169,59 @@ public static ITransformer BuildAndTrainModel(MLContext mlContext, IDataView spl
...
 
-You train the model, <xref:Microsoft.ML.Data.TransformerChain%601>, based on the dataset that has been loaded and transformed. Once the estimator has been defined, you train your model using the <xref:Microsoft.ML.Data.EstimatorChain%601.Fit%2A> method while providing the already loaded training data. This returns a model to use for predictions. `pipeline.Fit()` trains the pipeline and returns a `Transformer` based on the `DataView` passed in. The experiment is not executed until the `.Fit()` method runs.
+The [SdcaLogisticRegressionBinaryTrainer](xref:Microsoft.ML.Trainers.SdcaLogisticRegressionBinaryTrainer) is your classification training algorithm. This is appended to the `estimator` and accepts the featurized `SentimentText` (`Features`) and the `Label` input parameters to learn from the historic data.
 
-Add the following code to the `BuildAndTrainModel` method:
+## Train the model
+
+Fit the model to the `Train` data and return the trained model by adding the following as the next line of code in the `BuildAndTrainModel()` method:
```
Comment (luisquintanilla) [inline docs/machine-learning/tutorials/sentiment-analysis.md]: Not sure if Train needs to be inlined.

**Review by luisquintanilla — CHANGES_REQUESTED** [review]
Look good. Minor changes.

File: docs/machine-learning/tutorials/sentiment-analysis.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (JRAlexander) [reply docs/machine-learning/tutorials/sentiment-analysis.md]: Just defined them at first usage.

File: docs/machine-learning/tutorials/sentiment-analysis.md (hunk 6 above)
Comment (JRAlexander) [reply docs/machine-learning/tutorials/sentiment-analysis.md]: Its xreffed in the previous paragraph.

File: docs/machine-learning/tutorials/sentiment-analysis.md [hunk 8]
```diff
@@ -145,16 +84,27 @@ Remove the existing class definition and add the following code, which has two c
```
[the remark this answers is not in the document]
Comment (JRAlexander) [reply docs/machine-learning/tutorials/sentiment-analysis.md]: They should be gone. Not needed. Gone now!

**Review by luisquintanilla — APPROVED** [review]

File: docs/machine-learning/tutorials/sentiment-analysis.md [hunk 9]
```diff
@@ -347,21 +297,23 @@ Add a call to the new method from the `Main` method, right under the `Evaluate`
...
 Add a comment to test the trained model's prediction in the `Predict` method by creating an instance of `SentimentData`:
 
 [!code-csharp[PredictionData](~/samples/machine-learning/tutorials/SentimentAnalysis/Program.cs#CreateTestIssue1 "Create test data for single prediction")]
 
- You can use that to predict the positive or negative sentiment of a single instance of the comment data. To get a prediction, use <xref:Microsoft.ML.PredictionEngine%602.Predict%2A> on the data. Note that the input data is a string and the model includes the featurization. Your pipeline is in sync during training and prediction. You didn’t have to write preprocessing/featurization code specifically for predictions, and the same API takes care of both batch and one-time predictions.
+Pass the test comment data to the `Prediction Engine` by adding the following as the next lines of code in the `CallUseModelWithSingleItem()` method:
 
 [!code-csharp[Predict](~/samples/machine-learning/tutorials/SentimentAnalysis/Program.cs#Predict "Create a prediction of sentiment")]
 
+The [Predict()](xref:Microsoft.ML.PredictionEngine%602.Predict%2A) function makes a prediction on a single column of data.
```
Comment (natke) [inline docs/machine-learning/tutorials/sentiment-analysis.md]: column -> row

**Review by natke — APPROVED** [review]
One minor comment. Other than that LGTM!

**Comment by JRAlexander** [comment]
Thanks, @luisquintanilla and @natke!!
